### PR TITLE
Restructure

### PR DESCRIPTION
**Pull request by TrevorJTClarke**

### CODE REVIEW

**Comment by TrevorJTClarke** [comment]
Running `npm i` ends with: `npm ERR! cb() never called!`

**Comment by TrevorJTClarke** [comment]
@taylorjdawson is the README.md file still valid?

**Comment by TrevorJTClarke** [comment]
@taylorjdawson looks like `npm test` throws a single error:
```
web3data › web3data should accept a blockchain id & add to headers

  1 test failed
  4 tests skipped
  2 uncaught exceptions

  contract › Successfully gets contract source code

  /Users/trevorclarke/Github/web3data/web3data-js/test/contract.test.js:63

   62:     let response = await t.context.web3data.contract.getSourceCode(TOKEN_ADDRESS);
   63:     t.is(response.status, 200)
   64: });

  Difference:

  - undefined
  + 200
```
Otherwise, i like the progress on here. Let's discuss before we merge

**Comment by taylorjdawson** [comment]
I haven't added much to this branch I still need to do two things:
1. A while back I got `ava-playback` to work [by implementing a fix](https://github.com/dempfi/ava-playback/pull/14) - right now ava-playback doesn't work without my forked version however, the deps [points to a local copy](https://github.com/web3data/web3data-js/blob/a5b71b65f9d324df20115cae5a0c4a294b696dad/package.json#L56)
2. All tests need to be updated for endpoint changes, and re-recorded with ava-playback
3. That specific error is Endpoint related. See [this slack comment](https://amberdata.slack.com/archives/C5T75GEJ3/p1562343521087300)

**Comment by taylorjdawson** [comment]
> @taylorjdawson is the README.md file still valid?

It's outdated now.

**Comment by TrevorJTClarke** [comment]
First install fails for multiple reasons, this needs a bit of work

**Comment by TrevorJTClarke** [comment]
Error: Could not resolve required module 'ava-playback'

**Comment by TrevorJTClarke** [comment]
```
53 tests passed
  6 tests skipped
  2 uncaught exceptions

  Uncaught exception in test/websocket.test.js

  /Users/trevorclarke/Github/web3data/web3data-js/node_modules/ws/lib/websocket-server.js:76



  new WebSocketServer (node_modules/ws/lib/websocket-server.js:76:20)
  test/websocket.test.js:123:21



  Uncaught exception in test/websocket.test.js

  /Users/trevorclarke/Github/web3data/web3data-js/node_modules/ws/lib/websocket-server.js:76



  new WebSocketServer (node_modules/ws/lib/websocket-server.js:76:20)
  test/websocket.test.js:251:17
```

I was using `npm run test-live` with an ENT key

**Comment by TrevorJTClarke** [comment]
> > @taylorjdawson is the README.md file still valid?
> 
> It's outdated now.

@taylorjdawson we need to update README.md as well as package.json for this update.

**Comment by TrevorJTClarke** [comment]
> ```
> 53 tests passed
>   6 tests skipped
>   2 uncaught exceptions
> 
>   Uncaught exception in test/websocket.test.js
> 
>   /Users/trevorclarke/Github/web3data/web3data-js/node_modules/ws/lib/websocket-server.js:76
> 
> 
> 
>   new WebSocketServer (node_modules/ws/lib/websocket-server.js:76:20)
>   test/websocket.test.js:123:21
> 
> 
> 
>   Uncaught exception in test/websocket.test.js
> 
>   /Users/trevorclarke/Github/web3data/web3data-js/node_modules/ws/lib/websocket-server.js:76
> 
> 
> 
>   new WebSocketServer (node_modules/ws/lib/websocket-server.js:76:20)
>   test/websocket.test.js:251:17
> ```
> 
> I was using `npm run test-live` with an ENT key

@taylorjdawson im still getting this issue. NOTE: I added ".skip" to the contract get source code, since its not relevent for this PR. Thats when im getting issues on these tests

File: index.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const Web3Data = require('./src/web3data')
+// eslint-disable-next-line no-undef
+if (typeof window !== 'undefined' && typeof window.Web3Data === 'undefined') {
```
Comment (TrevorJTClarke) [inline index.js]: @taylorjdawson this usually is generated or setup differently, please see:
https://github.com/ethers-io/ethers.js/blob/master/dist/ethers.js#L1

**Pull request merged**